op-node/p2p/node.go

```diff
@@ -85,86 +110,83 @@ func (n *NodeP2P) init(resourcesCtx context.Context, rollupCfg *rollup.Config, l
 		return fmt.Errorf("failed to start p2p host: %w", err)
 	}
 
-	// TODO(CLI-4016): host is not optional, NodeP2P as a whole is. This if statement is wrong
-	if n.host != nil {
-		// Enable extra features, if any. During testing we don't setup the most advanced host all the time.
-		if extra, ok := n.host.(ExtraHostFeatures); ok {
-			n.gater = extra.ConnectionGater()
-			n.connMgr = extra.ConnectionManager()
-		}
-		eps, ok := n.host.Peerstore().(store.ExtendedPeerstore)
-		if !ok {
-			return fmt.Errorf("cannot init without extended peerstore: %w", err)
-		}
-		n.store = eps
-		scoreParams := setup.PeerScoringParams()
+	// Enable extra features, if any. During testing we don't setup the most advanced host all the time.
```

Comment:
The host check is moved out, I completed the reliance on SetupP2P.Disabled elsewhere as it looks like was intended.